### PR TITLE
Update ci images for Kokkos and Ginkgo

### PR DESCRIPTION
**Pull request by davidscn**
Related to https://github.com/precice/precice/pull/2051

### CODE REVIEW

File: ginkgo/ginkgo-install.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-VERSION=3.12.4
+VERSION=1.8.0
 
-wget -c https://github.com/ginkgo-project/ginkgo/archive/refs/heads/public_common_kernels.zip
-unzip public_common_kernels.zip && mv ginkgo-public_common_kernels ginkgo
+wget -c https://github.com/ginkgo-project/ginkgo/archive/refs/tags/v$VERSION.zip
+unzip v$VERSION.zip && mv ginkgo-$VERSION ginkgo
 cd ginkgo && mkdir build && cd build
-cmake .. && make
+cmake -DBUILD_TESTING="OFF" -DGINKGO_BUILD_EXAMPLES="OFF" -DGINKGO_BUILD_TESTS="OFF" .. && make
```
Comment (fsimonis) [inline ginkgo/ginkgo-install.sh:7]: Using more jobs should help reducing the compile times too. 

```suggestion
cmake -DBUILD_TESTING="OFF" -DGINKGO_BUILD_EXAMPLES="OFF" -DGINKGO_BUILD_TESTS="OFF" .. && make -j 4
```

**Review by fsimonis — APPROVED** [review]
Looks good.

This means that the Ginkgo tests will migrate from Ubuntu to Fedora. Currently open PRs are expected fail.

File: ginkgo/ginkgo-install.sh (hunk 1 above)
Comment (davidscn) [reply ginkgo/ginkgo-install.sh:7]: I don't think that the GitHub runners have four cores available, but ok

Comment (fsimonis) [reply ginkgo/ginkgo-install.sh]: Github upgraded its runners some time ago

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Pull request merged**